kernel/arch/dreamcast/include/dc/maple/controller.h

```diff
@@ -223,7 +223,8 @@ typedef void (*cont_btn_callback_t)(uint8_t addr, uint32_t btns);
     \param  addr            The controller to listen on (or 0 for all ports). 
                             This value can be obtained by using maple_addr().
     \param  btns            The buttons bitmask to match.
-    \param  cb              The callback to call when the buttons are pressed.
+    \param  cb              The callback to call when the buttons are pressed, 
+                            or NULL to un-set the current callback.
```

Comment:
The part above this in the comment probably should get updated, since it says that the callback will be called in an interrupt context, which is no longer true after this PR is merged.

Also, it'd probably be a good idea to also note that only one of these can be set up, since otherwise a user might think they can schedule as many as they want (I seem to remember at least one topic on DCEmulation where someone was trying to make one of these per controller or something before).

Reply:
Updated the notes on this with both parts. I forgot when I fast-forwarded that Falco had added in the notes about the interrupt context.